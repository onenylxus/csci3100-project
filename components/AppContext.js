// Import
import React from 'react';

// Export authentication context
export default React.createContext();
