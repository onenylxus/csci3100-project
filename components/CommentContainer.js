// Import
import React from 'react';
import { View, ScrollView } from 'react-native';
import CreateCommentForm from './CreateCommentForm';
import Comment from './Comment';

// Export
export default function CommentContainer({ post, state }) {
  const [list, setList] = React.useState([]);

  const status = React.useRef(0);
  const postId = React.useRef(post._id);

  function fetchComment() {
    (async () => {
      await fetch('https://cu-there-server.herokuapp.com/fetchComment', {
        method: 'POST',
        headers: {
          'Content-Type': 'application/json',
        },
        body: JSON.stringify({
          _id: postId.current,
        }),
      })
        .then((res) => {
          status.current = res.status;
          return res;
        })
        .then((res) => res.json())
        .then((res) => {
          if (status.current === 200) {
            setList(res.comment);
          } else if (status.current === 422) {
            console.log(res.error);
          }
        })
        .catch((err) => console.log(err));
    })();
  }

  React.useEffect(fetchComment, [state]);

  function generate() {
    return list.map((comment) => (
      <Comment key={comment.postId} comment={comment} />
    ));
  }

  return (
    <View style={{ marginBottom: 50 }}>
      <ScrollView>
        <View>{generate()}</View>
        <CreateCommentForm post={post} />
      </ScrollView>
    </View>
  );
}
