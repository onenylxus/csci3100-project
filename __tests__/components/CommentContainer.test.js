// Import
import React from 'react';
import { render } from '@testing-library/react-native';

// Import target component
import AppContext from '../../components/AppContext';
import CommentContainer from '../../components/CommentContainer';

// Mock authentication method
const AppMethodMock = {
  login: jest.fn(),
  logout: jest.fn(),
  askPerm: jest.fn(),
  getUser: jest.fn(),
  getPlatform: jest.fn(),
  getCameraPerm: jest.fn(),
  getImagePerm: jest.fn(),
};

// Mock FontAwesome icons
jest.mock('@fortawesome/react-native-fontawesome', () => ({
  FontAwesomeIcon: '',
}));

// Run jest tests
describe('CommentContainer', () => {
  let element;

  beforeEach(() => {
    // Mock post
    const postMock = {
      username: 'testac',
      timestamp: new Date(2021, 4, 1),
      title: '',
      content: '',
      peopleLike: [],
      peopleDislike: [],
    };

    // Render
    element = render(
      <AppContext.Provider value={AppMethodMock}>
        <CommentContainer post={postMock} />
      </AppContext.Provider>
    );
  });

  it('matches snapshot', () => {
    expect(element.toJSON()).toMatchSnapshot();
  });
});
