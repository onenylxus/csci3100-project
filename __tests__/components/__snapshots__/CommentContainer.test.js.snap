// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CommentContainer matches snapshot 1`] = `
<View>
  <RCTScrollView>
    <View>
      <View />
      <View
        style={
          Object {
            "alignSelf": "center",
            "backgroundColor": "#ffffff",
            "borderColor": "#000000",
            "borderRadius": 15,
            "borderWidth": 0.5,
            "flexDirection": "row",
            "height": 40,
            "margin": "4%",
            "width": "85%",
          }
        }
      >
        <
          icon={
            Object {
              "icon": Array [
                512,
                512,
                Array [],
                "f304",
                "M290.74 93.24l128.02 128.02-277.99 277.99-114.14 12.6C11.35 513.54-1.56 500.62.14 485.34l12.7-114.22 277.9-277.88zm207.2-19.06l-60.11-60.11c-18.75-18.75-49.16-18.75-67.91 0l-56.55 56.55 128.02 128.02 56.55-56.55c18.75-18.76 18.75-49.16 0-67.91z",
              ],
              "iconName": "pen",
              "prefix": "fas",
            }
          }
          size={15}
          style={
            Object {
              "marginHorizontal": 5,
              "marginVertical": 12,
            }
          }
        />
        <TextInput
          allowFontScaling={true}
          onChangeText={[Function]}
          placeholder="Write Comment"
          rejectResponderTermination={true}
          style={
            Object {
              "width": "75%",
            }
          }
          underlineColorAndroid="transparent"
          value=""
        />
        <View
          accessible={true}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignContent": "center",
              "justifyContent": "center",
              "marginHorizontal": 5,
              "opacity": 1,
            }
          }
        >
          <Text
            style={
              Object {
                "color": "blue",
              }
            }
          >
            Send
          </Text>
        </View>
      </View>
    </View>
  </RCTScrollView>
</View>
`;
